src/schema/posts.ts

```diff
@@ -892,6 +913,15 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    checkLinkPreview: async (
+      _,
+      { url, sourceId }: SubmitExternalLinkArgs,
+      ctx,
+    ): Promise<ExternalLinkPreview> => {
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
+
+      return fetchLinkPreview(url);
```

Comment:
Would probably await `fetchLinkPreview` here and make sure only fields we expect (`ExternalLinkPreview`) are returned from the `checkLinkPreview`, if API changes in any way there could be some additional fields leaking unnecessarily.

Reply:
This would be bound to the GraphQL schema for the type we return. It would only allow returning those properties we define on the endpoint. So when there are additional ones, they won't be provided unnecessarily.